doc/dev/debugging.rst

```diff
@@ -388,3 +388,84 @@ corresponding arguments.
     $ /path/to/executable
     MemRef: base@ = 0x64fc9dd5ffc0 rank = 0 offset = 0 sizes = [] strides = [] data =
     25
+
+Mixed-mode debugging of Python and C++
+======================================
+
+Catalyst supports mixed-mode debugging of Python and/or C++ code when providing the ``debug_compiler=True`` flag to 
+the ``@qjit`` decorator. Enabling this option signals to the compiler to wait for an appropriate user-provided signal
+after launching the compiler process. Some notes about use of this support:
+
+* This functionality requires building Catalyst with debug symbols. This can be achieved via make all 
+  ``BUILD_TYPE="RelWithDebInfo"``. The debug symbols are only within available within the Catalyst-owned targets. 
+To enable debugging of LLVM and other associated external libraries and binaries, ensure the 
+  ``BUILD_TYPE_EXT="RelWithDebInfo"`` option is also set.
+* Launching the C++ debugger requires attaching to a running process. This often requires ``sudo`` privileges on the 
+  running system.
+* The spawned compiler subprocess immediately issues a ``SIGSTOP`` signal to avoid execution of the compiler. To 
+  continue execution requires receipt of a ``SIGCONT`` signal after the C++ debugger has attached.
+* To validate if running within an active (Python) debugger session, the function :func:`~.debug.debugger.is_debugger_active`
+  can be used.
+
+The signalling steps can be provided via an active terminal session as
+
+.. code-block:: shell
+
+    $ kill -s SIGCONT <PID>
+
+where ``<PID>`` is the process-ID. This can also be issued from an active Python debugger session, such as through VSCode's 
+debug terminal as
+
+.. code-block:: python
+
+    import os, signal
+    os.kill(<PID>, signal.SIGCONT)
+
+To enable support from VSCode, the following configuration files can be used to add debugger configurations for Python, and 
+C++.
```

Comment:
I'm trying to test this out in vscode, but I don't actually fully understand how everything is supposed to work 😅 
Maybe I can ask you some questions in person next week? :)

Reply:
Yea, the internal guide is currently the source-of-truth on set-up here. We can discuss next week and see what is portable across to here.